rinad/src/ipcp/ipc-process.cc

```diff
@@ -654,372 +663,212 @@ void IPCProcessImpl::processFwdCDAPMsgEvent(
-	ipcp->requestPDUFTEDump();
-}
-
-static void
-application_registration_request_event_handler(rina::IPCEvent *e,
-		EventLoopData *opaque)
-{
-	DOWNCAST_DECL(e, rina::ApplicationRegistrationRequestEvent, event);
-	DOWNCAST_DECL(opaque, IPCProcessImpl, ipcp);
-
-	ipcp->namespace_manager_->processApplicationRegistrationRequestEvent(*event);
-}
-
-static void
-application_unregistration_request_event_handler(rina::IPCEvent *e,
-		EventLoopData *opaque)
-{
-	DOWNCAST_DECL(e, rina::ApplicationUnregistrationRequestEvent, event);
-	DOWNCAST_DECL(opaque, IPCProcessImpl, ipcp);
-
-	ipcp->namespace_manager_->processApplicationUnregistrationRequestEvent(*event);
-}
-
-static void
-ipc_process_create_connection_response_handler(rina::IPCEvent *e,
-		EventLoopData *opaque)
-{
-	DOWNCAST_DECL(e, rina::CreateConnectionResponseEvent, event);
-	DOWNCAST_DECL(opaque, IPCProcessImpl, ipcp);
-
-	ipcp->flow_allocator_->processCreateConnectionResponseEvent(*event);
-}
-
-static void
-allocate_flow_response_event_handler(rina::IPCEvent *e,
-		EventLoopData *opaque)
-{
-	DOWNCAST_DECL(e, rina::AllocateFlowResponseEvent, event);
-	DOWNCAST_DECL(opaque, IPCProcessImpl, ipcp);
-
-	ipcp->flow_allocator_->submitAllocateResponse(*event);
-}
-
-static void
-ipc_process_create_connection_result_handler(rina::IPCEvent *e,
-		EventLoopData *opaque)
-{
-	DOWNCAST_DECL(e, rina::CreateConnectionResultEvent, event);
-	DOWNCAST_DECL(opaque, IPCProcessImpl, ipcp);
-
-	ipcp->flow_allocator_->processCreateConnectionResultEvent(*event);
-}
-
-static void
-ipc_process_update_connection_response_handler(rina::IPCEvent *e,
-		EventLoopData *opaque)
-{
-	DOWNCAST_DECL(e, rina::UpdateConnectionResponseEvent, event);
-	DOWNCAST_DECL(opaque, IPCProcessImpl, ipcp);
-
-	ipcp->flow_allocator_->processUpdateConnectionResponseEvent(*event);
-}
-
-static void
-flow_deallocation_requested_event_handler(rina::IPCEvent *e,
-		EventLoopData *opaque)
-{
-	DOWNCAST_DECL(e, rina::FlowDeallocateRequestEvent, event);
-	DOWNCAST_DECL(opaque, IPCProcessImpl, ipcp);
-
-	ipcp->flow_allocator_->submitDeallocate(*event);
-}
-
-static void
-ipc_process_destroy_connection_result_handler(rina::IPCEvent *e,
-		EventLoopData *opaque)
-{
-	DOWNCAST_DECL(e, rina::DestroyConnectionResultEvent, event);
-
-	if (event->result != 0){
-		LOG_IPCP_WARN("Problems destroying connection with associated to port-id %d",
-				event->portId);
+void IPCProcessImpl::event_loop(void){
+
+	rina::IPCEvent *e;
+
+	bool keep_running = true;
+
+	LOG_DBG("Starting main I/O loop...");
+
+	while(keep_running) {
+		e = rina::ipcEventProducer->eventTimedWait(
+						IPCP_EVENT_TIMEOUT_S,
+						IPCP_EVENT_TIMEOUT_NS);
+		if(!e)
+			continue;
+
+		if(!keep_running)
+			break;
+
+		LOG_IPCP_DBG("Got event of type %s and sequence number %u",
+							rina::IPCEvent::eventTypeToString(e->eventType).c_str(),
+							e->sequenceNumber);
+
+		switch(e->eventType){
+			case rina::IPC_PROCESS_DIF_REGISTRATION_NOTIFICATION:
```

Comment:
Can we reduce a bit the indentation here, to keep things on 80 cols? I think we can align "switch" and "case" at least.